src/settings/CirculationRules.js

```diff
@@ -187,7 +187,9 @@ class CirculationRules extends React.Component {
 
   onSubmit(values) {
     const rules = values.rules.replace(/\t/g, '    ');
+
     this.save(rules);
+    this.setState({ ...values });
```

Comment:
```suggestion
    this.setState({ rules });
```
Should it be like that? As for now, you are saving all properties to the state.